cmd/registry-replacer/main.go

```diff
@@ -189,6 +202,18 @@ type usernameToken struct {
 	token    string
 }
 
+func configResolver(info *config.Info, resolverAddress, registryPath string) (*api.ReleaseBuildConfiguration, error) {
```

Comment:
This means that we have to do a http call for every single config. I would prefer it if we instead just load the config from the filesystem (we operate on o/release anyways), you can use the RegistryAgent constructed through pkg/load/agents.NewRegistryAgent for that and then use its `ResolveConfig`

Reply:
Ok, that makes sense. I wasn't sure if the `registry-replacer` would always have access to the registry locally but if it does it definitely makes sense to avoid having to make an unnecessary http call.

Reply:
We run it as part of the autoconfigbrancher tool which is essentially a tool that runs a bunch of other tools to somehow fix config periodically and then creates an automerging PR. Initially it was for branching, nowadays for all kinds of things: https://github.com/openshift/ci-tools/blob/d861eedff2eb28b479d20d465f9b670d3097691d/cmd/autoconfigbrancher/main.go#L157

The job that runs it has both the openshift/release and the ocp-build-data repo: https://github.com/openshift/release/blob/425ad253347fda93e6079dcc148c3e865463d917/ci-operator/jobs/infra-periodics.yaml#L631-L637